### PR TITLE
running in impractical footwear makes you trip

### PR DESCRIPTION
**Pull request by alsoandanswer**
you can't run in heels you know.

technically heels can eyestab so consider this a balance as well

### CODE REVIEW

**Comment by alsoandanswer** [comment]
!review

**Comment by TheDanseMacabre** [comment]
Hilariously based PR

File: code/modules/clothing/shoes/jobs.dm [hunk 1]
```diff
@@ -28,12 +28,18 @@
 	icon_state = "kneeboots"
 	item_state = "kneeboots"
 
+/obj/item/clothing/shoes/jackboots/knee/handle_movement(var/turf/walking, var/running)
+	trip_up()
```
Comment (MarinaGryphon) [inline code/modules/clothing/shoes/jobs.dm:32]: These aren't heels, they're boots that reach higher. There is no conceivable reason you would trip while wearing them.

File: code/modules/clothing/shoes/jobs.dm [hunk 2]
```diff
@@ -28,12 +28,18 @@
...
+
 /obj/item/clothing/shoes/jackboots/thigh
 	name = "thigh-length black boots"
 	desc = "Even taller synthleather boots with an artificial shine."
 	icon_state = "thighboots"
 	item_state = "thighboots"
 
+/obj/item/clothing/shoes/jackboots/thigh/handle_movement(var/turf/walking, var/running)
+	trip_up()
```
Comment (MarinaGryphon) [inline code/modules/clothing/shoes/jobs.dm:41]: Ditto. Not heels.

File: code/modules/clothing/shoes/miscellaneous.dm [hunk 3]
```diff
@@ -195,6 +194,9 @@ obj/item/clothing/shoes/sandal/clogs
 		M = user
 	return eyestab(M,user)
 
+/obj/item/clothing/shoes/heels/handle_movement(var/turf/walking, var/running)
+	trip_up()
```
Comment (MarinaGryphon) [inline code/modules/clothing/shoes/miscellaneous.dm:198]: You didn't pass the arguments to the proc, so it wouldn't work anyway.

**Review by MarinaGryphon — CHANGES_REQUESTED** [review]
I forgot to click request changes.

File: code/modules/clothing/shoes/jobs.dm (hunk 2 above)
Comment (alsoandanswer) [reply code/modules/clothing/shoes/jobs.dm:41]: yeah nah went and looked at a google search and its all heels

also having boots laced up to your thighs is probably a bad idea to run in

File: code/modules/clothing/shoes/jobs.dm (hunk 1 above)
Comment (alsoandanswer) [reply code/modules/clothing/shoes/jobs.dm:32]: i'll give you this one

**Review by Arrow768 — APPROVED** [review]

**Review by skull132 — DISMISSED** [review]
Only point of concern: 25% per movement is a hilariously high chance of tripping. It'll basically make running in that kind of footwear a meme, with you tripping very quickly. I'd suggest making it way, way smaller of a chance, so that you can actually get a decent distance going before planting your face on the floor.

**Comment by NonQueueingMatt** [comment]
Agreed with Skull. 2% chance should be enough, really.

File: code/modules/clothing/clothing.dm [hunk 4]
```diff
@@ -752,6 +753,18 @@
...
+	if(!running)
+		if(footstep >= 2)
+			footstep = 0
+		else
+			footstep++
+	else
+		if(prob(25))
+			if(ismob(usr))
+				var/mob/M = usr
+				M.Weaken(2)
```
Comment (NonQueueingMatt) [inline code/modules/clothing/clothing.dm:766]: `usr` in procs is not something we want to use, generally. It's unsafe, therefore we should ask ourselves here - how do we do this in a safer fashion?

The simpler solution is just to check the loc of the shoes - which should always be a mob, and if it is, trip 'em.

Also add a warning message so people know why they're tripping.

**Review by NonQueueingMatt — DISMISSED** [review]

File: code/modules/clothing/clothing.dm (hunk 4 above)
Comment (alsoandanswer) [reply code/modules/clothing/clothing.dm:766]: Fixed

**Review by NonQueueingMatt — APPROVED** [review]

**Pull request merged**